File: app/src/main/java/com/example/pam_project/database/tasks/TaskDao.java
package com.example.pam_project.database.tasks;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import io.reactivex.Flowable;

@Dao
public interface TaskDao {
    @Query("SELECT * FROM Tasks WHERE id =:id")
    Flowable<TaskEntity> getTaskById(final long id);

    @Insert
    void insertTask(final TaskEntity task);

    @Update
    void updateTask(final TaskEntity task);

    @Delete
    void deleteTask(final TaskEntity task);
}

File: app/src/main/java/com/example/pam_project/database/tasks/TaskDao.kt
package com.example.pam_project.database.tasks

import androidx.room.*
import io.reactivex.Flowable

@Dao
interface TaskDao {
    @Query("SELECT * FROM Tasks WHERE id =:id")
    fun getTaskById(id: Long): Flowable<TaskEntity?>

    @Insert
    fun insertTask(task: TaskEntity?)

    @Update
    fun updateTask(task: TaskEntity?)

    @Delete
    fun deleteTask(task: TaskEntity?)
}